src/common/memory_desc_wrapper.hpp

```diff
@@ -424,31 +425,38 @@ struct memory_desc_wrapper : public c_compatible {
 };
 
 inline bool memory_desc_wrapper::similar_to(const memory_desc_wrapper &rhs,
-        bool with_padding, bool with_data_type, int dim_start) const {
+        bool with_padding, bool with_data_type, int dim_start, int stride_start, bool is_weak_cmp, bool check_off0) const {
     using namespace utils;
 
     if (one_of(format_kind(), format_kind::undef, format_kind::any))
         return false;
     if (is_wino_desc() || is_rnn_packed_desc()) return false;
 
     const int ds = dim_start;
+    if (stride_start == -1)
+        stride_start = ds;
     const auto &blk = blocking_desc();
     const auto &r_blk = rhs.blocking_desc();
 
+    bool isSameDims = is_weak_cmp ? array_cmp_weak(dims() + ds, rhs.dims() + ds, ndims() - ds)
+                        : array_cmp(dims() + ds, rhs.dims() + ds, ndims() - ds);
```

Comment:
```suggestion
    auto custom_cpm = is_weak_cmp ? array_cmp_weak : array_cmp;
    bool isSameDims = custom_cpm (dims() + ds, rhs.dims() + ds, ndims() - ds);
```
    Then you can inline all the `isSame` back again. The code will look more compact. What do you think?